src/components/CommentsList.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Comment, CommentData } from '../types/Comment';
+import { Loader } from './Loader';
+import { NewCommentForm } from './NewCommentForm';
+import { deleteComment, getComments, postComment } from '../api/comments';
+
+type Props = {
+  postId: number;
+};
+
+export const CommentsList: React.FC<Props> = ({ postId }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadError, setLoadError] = useState(false);
+  const [addError, setAddError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(false);
+
+  const getPostCommentsFromServer = async () => {
+    try {
+      setComments([]);
+      setIsFormVisible(false);
+      setLoadError(false);
+      setIsLoading(true);
+      const commentsFromServer = await getComments(postId);
+
+      setComments(commentsFromServer);
+    } catch {
+      setLoadError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const postNewComment = useCallback(async (comment: CommentData) => {
+    const newComment = { ...comment, postId };
+
+    try {
+      const addedComment = await postComment(newComment);
+
+      setComments(currentComment => [...currentComment, addedComment]);
+    } catch {
+      setAddError(true);
+    }
+  }, []);
+
+  const handleDeleteComment = useCallback(async (commentId: number) => {
+    const tempComments = [...comments];
+
+    setComments(currentComments => (
+      currentComments.filter(comment => comment.id !== commentId)
+    ));
+
+    try {
+      setDeleteError(false);
+      await deleteComment(commentId);
+    } catch {
+      setComments(tempComments);
+      setDeleteError(true);
+    }
+  }, []);
+
+  useEffect(() => {
+    getPostCommentsFromServer();
```

Comment:
Consider moving the 'getPostCommentsFromServer' call inside of the useEffect to avoid dependency warning.